src/arche/rules/result.py

```diff
@@ -80,6 +94,25 @@ def info(self):
     def warnings(self):
         return self.messages.get(Level.WARNING)
 
+    @property
+    def outcome(self) -> Outcome:
+        if not self._outcome:
```

Comment:
These conditions could go in a separate method, so it would be clear they are getting an `Outcome` and setting `_outcome` if it is not set yet.

Reply:
What happens if I set an `outcome`, say `PASSED`, but added an error later?

Reply:
My idea is to set outcome based on messages' levels and allow to override it.
If `_outcome` has the value, it'll returned notwithstanding what else is in messages.

Reply:
@ejulio I agree with @manycoding. There's the default behavior of defining the outcome based on the message log levels; In the case that the rule has some exception and the outcome can't be defined based on the messages alone, the rule can override it.